policy/modules/services/container.te

```diff
@@ -475,6 +527,36 @@ logging_send_audit_msgs(container_t)
 
 userdom_use_user_ptys(container_t)
 
+tunable_policy(`container_use_host_all_caps',`
+	# omitted sys_module
+	allow container_t self:capability { chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap };
+	# omitted mac_admin, mac_override
+	allow container_t self:capability2 { syslog wake_alarm block_suspend audit_read perfmon bpf checkpoint_restore };
+')
+
+tunable_policy(`container_use_host_mknod',`
+	allow container_t self:capability mknod;
+')
+
+tunable_policy(`container_use_host_sysadmin',`
+	allow container_t self:capability sys_admin;
+')
+
+tunable_policy(`container_use_userns_all_caps',`
+	# omitted sys_module
+	allow container_t self:cap_userns { chown dac_override dac_read_search fowner fsetid kill setgid setuid setpcap linux_immutable net_bind_service net_broadcast net_admin net_raw ipc_lock ipc_owner sys_rawio sys_chroot sys_ptrace sys_pacct sys_admin sys_boot sys_nice sys_resource sys_time sys_tty_config mknod lease audit_write audit_control setfcap };
+	# omitted mac_admin, mac_override
+	allow container_t self:cap2_userns { syslog wake_alarm block_suspend audit_read perfmon bpf checkpoint_restore };
+')
+
+tunable_policy(`container_use_userns_mknod',`
+	allow container_t self:cap_userns mknod;
+')
+
+tunable_policy(`container_use_userns_sysadmin',`
+	allow container_t self:cap_userns sys_admin;
+')
+
```

Comment:
Any reason to only have broken-out `mknod` and `sys_admin`? (I'm not asking for a tunable for each cap)

One thing to consider would be that the host cap could possibly imply the `ns` version. e.g. line 556 could be `container_use_userns_sysadmin | container_use_host_sysadmin`.

Reply:
> Any reason to only have broken-out `mknod` and `sys_admin`? (I'm not asking for a tunable for each cap)

This is just how container-selinux handles it and I thought their implementation of this logic made sense.

> One thing to consider would be that the host cap could possibly imply the `ns` version. e.g. line 556 could be `container_use_userns_sysadmin | container_use_host_sysadmin`.

I think that's a great idea -- I don't think there's much reason/functionality to be gained from allowing a capability on the host but not within the user namespace.

Reply:
> > Any reason to only have broken-out `mknod` and `sys_admin`? (I'm not asking for a tunable for each cap)
> 
> This is just how container-selinux handles it and I thought their implementation of this logic made sense.

Ok, I'm fine with that.